sky/engine/tonic/dart_message_handler.h

```diff
@@ -0,0 +1,51 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SKY_ENGINE_TONIC_DART_MESSAGE_HANDLER_H_
+#define SKY_ENGINE_TONIC_DART_MESSAGE_HANDLER_H_
+
+#include "base/callback.h"
+#include "base/message_loop/message_loop.h"
+#include "dart/runtime/include/dart_api.h"
+
+namespace blink {
+class DartState;
+
+class DartMessageHandler {
+ public:
+  DartMessageHandler();
+  virtual ~DartMessageHandler();
+
+  // Messages for the current isolate will be scheduled on |runner|.
+  void Initialize(const scoped_refptr<base::SingleThreadTaskRunner>& runner);
+
+ protected:
+  // Called from an unknown thread for each message.
+  virtual void OnMessage(DartState* dart_state);
+  // By default, called on the task runner's thread for each message.
+  virtual void OnHandleMessage(DartState* dart_state);
```

Comment:
@abarth Suggested during mojo review to allow someone to override the behaviour.